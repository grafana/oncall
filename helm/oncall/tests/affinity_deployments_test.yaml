suite: test image and imagePullPolicy for deployments
templates:
  - celery/deployment.yaml
  - engine/deployment.yaml
  - engine/job-migrate.yaml
release:
  name: oncall
tests:
  - it: affinity={} -> should set afffinity null
    asserts:
      - notExists:
          path: spec.template.spec.affinity

  - it: affinity -> should use custom affinity
    set:
      migrate:
        affinity:
          podAntiAffinity:
            preferredDuringSchedulingIgnoredDuringExecution:
              - weight: 100
                podAffinityTerm:
                  labelSelector:
                    matchExpressions:
                      - key: app.kubernetes.io/name
                        operator: In
                        values:
                          - grafana
                      - key: app.kubernetes.io/instance
                        operator: In
                        values:
                          - grafana
                  topologyKey: "failure-domain.beta.kubernetes.io/zone"
      engine:
        affinity:
          podAntiAffinity:
            preferredDuringSchedulingIgnoredDuringExecution:
              - weight: 100
                podAffinityTerm:
                  labelSelector:
                    matchExpressions:
                      - key: app.kubernetes.io/name
                        operator: In
                        values:
                          - grafana
                      - key: app.kubernetes.io/instance
                        operator: In
                        values:
                          - grafana
                  topologyKey: "failure-domain.beta.kubernetes.io/zone"
      celery:
        affinity:
          podAntiAffinity:
            preferredDuringSchedulingIgnoredDuringExecution:
              - weight: 100
                podAffinityTerm:
                  labelSelector:
                    matchExpressions:
                      - key: app.kubernetes.io/name
                        operator: In
                        values:
                          - grafana
                      - key: app.kubernetes.io/instance
                        operator: In
                        values:
                          - grafana
                  topologyKey: "failure-domain.beta.kubernetes.io/zone"
    asserts:
      - matchSnapshot:
          path: spec.template.spec.affinity
