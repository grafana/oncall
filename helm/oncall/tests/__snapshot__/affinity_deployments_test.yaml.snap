affinity -> should use custom affinity:
  1: |
    podAntiAffinity:
      preferredDuringSchedulingIgnoredDuringExecution:
        - podAffinityTerm:
            labelSelector:
              matchExpressions:
                - key: app.kubernetes.io/name
                  operator: In
                  values:
                    - grafana
                - key: app.kubernetes.io/instance
                  operator: In
                  values:
                    - grafana
            topologyKey: failure-domain.beta.kubernetes.io/zone
          weight: 100
  2: |
    podAntiAffinity:
      preferredDuringSchedulingIgnoredDuringExecution:
        - podAffinityTerm:
            labelSelector:
              matchExpressions:
                - key: app.kubernetes.io/name
                  operator: In
                  values:
                    - grafana
                - key: app.kubernetes.io/instance
                  operator: In
                  values:
                    - grafana
            topologyKey: failure-domain.beta.kubernetes.io/zone
          weight: 100
  3: |
    podAntiAffinity:
      preferredDuringSchedulingIgnoredDuringExecution:
        - podAffinityTerm:
            labelSelector:
              matchExpressions:
                - key: app.kubernetes.io/name
                  operator: In
                  values:
                    - grafana
                - key: app.kubernetes.io/instance
                  operator: In
                  values:
                    - grafana
            topologyKey: failure-domain.beta.kubernetes.io/zone
          weight: 100
